src/db.rs

```diff
@@ -321,6 +328,14 @@ impl Group {
             }
         }
     }
+
+    /// Get a timestamp field by name
+    pub fn get_time(&self, key: &str) -> Option<&chrono::NaiveDateTime> {
+        match self.times.get(key) {
```

Comment:
Since `get` already should return an `Option<&T>`, you should just be able to write

```rust
pub fn get_time(&self, key: &str) { 
    self.times.get(key)
}
```

This might need some additional `.as_ref()` , `.as_deref()` , etc . if you run into type checking issues.

Reply:
I will see if I can make this work and either close this or comment further if I need assistance.

Reply:
I am part of the way there.  The following code works; but, the compiler cannot infer the return type of the method if I leave off the `-> Option<&chrono::NaiveDateTime>`. 

Are you thinking I should be able to eliminate this explicit return type declaration as well?

```
/// Get a timestamp field by name
pub fn get_time(&self, key: &str) -> Option<&chrono::NaiveDateTime> {
    self.times.get(key)
}
```

Reply:
Oh sorry, I messed up the example code. Rust always requires explicit return type annotations for function definitions.